packages/sources/src/addSourceWizard/SourceAddSchema.js

```diff
@@ -3,11 +3,27 @@ import { componentTypes, validatorTypes } from '@data-driven-forms/react-form-re
 import { Popover, TextContent, TextList, TextListItem, Text, TextVariants, Title } from '@patternfly/react-core';
 import { QuestionCircleIcon } from '@patternfly/react-icons';
 import SSLFormLabel from './SSLFormLabel';
-import { AwsIcon, OpenshiftIcon } from '@patternfly/react-icons';
+import { AwsIcon, OpenshiftIcon, MicrosoftIcon } from '@patternfly/react-icons';
+import debouncePromise from '../utilities/debouncePromise';
+import { findSource } from '../api';
+
+export const asyncValidator = (value, sourceId = undefined) => findSource(value).then(({ data: { sources }}) => {
+    if (sources.find(({ id }) => id !== sourceId)) {
+        return 'Name has already been taken';
+    }
+
+    if (value === '' || value === undefined) {
+        return 'Name can\'t be blank';
+    }
+
+    return undefined;
```

Comment:
Undefined is returned by default. So no need for it here.

Reply:
I think this is more readable. Like "it is our intention to return undefined and it's not a mistake."